src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCITriggerService.java

```diff
@@ -31,6 +31,11 @@ public void triggerBuild(ProgrammingExerciseParticipation participation) throws
         triggerBuild(participation.getVcsRepositoryUrl(), participation.getProgrammingExercise().getBranch());
     }
 
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation, String commitHash) throws ContinuousIntegrationException {
+        triggerBuild(participation.getVcsRepositoryUrl(), commitHash);
```

Comment:
Does this actually work, using the commit hash instead of the branch?

Reply:
It should be solved now. The method was removed as it was unnecessary.